sys/ktest.c

```diff
@@ -0,0 +1,175 @@
+#include <ktest.h>
+#include <stdc.h>
+#include <malloc.h>
+
+/* Borrowed from mips/malta.c */
+char *kenv_get(const char *key);
+
+/* Stores currently running test data. */
+static test_entry_t *current_test = NULL;
+/* A null-terminated array of pointers to the tested test list. */
+static test_entry_t **autorun_tests;
+
+int ktest_test_running_flag = 0;
+
+/* We only need this pool to allocate some memory for performing operations on
+ * the list of all tests. */
+static MALLOC_DEFINE(test_pool, "test data pool");
```

Comment:
I'd prefer the test framework not to rely on dynamic memory allocation, which is going to be test subject on its own. Could you introduce `test_entry_swap(a, b)` routine and perform `qsort` on the linker set with not-runnable tests move to the bottom of `tests` array?